src/plugins/chart_expressions/expression_partition_vis/public/utils/layers/get_color.ts

```diff
@@ -163,12 +186,14 @@ const overrideColors = (
 
 export const getColor = (
   chartType: ChartTypes,
-  d: ShapeTreeNode,
+  // FIXME this could be both a string or a RangeKey see https://github.com/elastic/kibana/issues/153437
+  categoricalKey: string, // could be RangeKey
+  arrayNode: SimplifiedArrayNode,
   layerIndex: number,
   isSplitChart: boolean,
   overwriteColors: { [key: string]: string } = {},
-  columns: Array<Partial<BucketColumns>>,
-  rows: DatatableRow[],
+  distinctSeries: DistinctSeries,
+  { columnsLength, rowsLength }: { columnsLength: number; rowsLength: number },
```

Comment:
instead of passing columns and rows, I thought it was way cleaner(and easier to test) passing only the information needed by the function without any other complexity